packages/material-ui-system/src/styleFunctionSx.js

```diff
@@ -1,4 +1,4 @@
-import { deepmerge } from '@material-ui/utils';
+import merge from './merge';
```

Comment:
`merge` is more suited and performant for these use-cases then `deepmerge` - it returns early for null args + doesn't clone the first arg.